pkg/plugin/vm_restore_item_action.go

```diff
@@ -63,8 +64,11 @@ func (p *VMRestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (
 		return nil, errors.WithStack(err)
 	}
 
-	for i := 0; i < len(vm.Spec.Template.Spec.Domain.Devices.Interfaces); i++ {
-		vm.Spec.Template.Spec.Domain.Devices.Interfaces[i].MacAddress = ""
+	if util.IsMacAdressClearedByAnnotation(vm) {
```

Comment:
typo Adress -> address
Also I think the name is missleading.. 
consider something like: ShouldClearMacAddress

Reply:
Related to my comment above